pytests/test_event_time.py

```diff
@@ -0,0 +1,77 @@
+from datetime import datetime, timedelta
+
+from bytewax.dataflow import Dataflow
+from bytewax.execution import run_main
+from bytewax.inputs import ManualInputConfig
+from bytewax.window import TumblingWindowConfig, EventClockConfig, TestingClockConfig
+from bytewax.outputs import TestingOutputConfig
+
+
+def test_event_time_processing():
+    """
+    Test used to validate the EventClockConfig workings.
+    """
+    start_at = datetime.now()
+    late = timedelta(seconds=10)
+    window_length = timedelta(seconds=5)
+
+    # Utility functions to generate "temp" events
+    def temp(val, seconds):
+        time = start_at + timedelta(seconds=seconds)
+        return None, {"type": "temp", "time": time, "value": val}
+
+    # Utility functions to generate wait events (used for advancing the clock)
+    def wait():
+        return None, {"type": "temp", "time": start_at}
+
+    def input_builder(worker_index, worker_count, state):
+        # Each yield advances the clock by 1 second
+        # This should be processed in the first window
+        yield temp(1, 1)
+        # This too should be processed in the first window
+        yield temp(2, 2)
+        # Wait 10 seconds
+        for _ in range(10):
```

Comment:
Hmm. Not for this PR. But do you think it would be easier if `TestingClock` instead had a `time_getter` and let you explicitly set the time in Python in the tests? I think this pattern of needing items to implicitly advance the clock is getting more confusing than clear. I might prototype this out in another PR.

Reply:
Here I think this will help: https://github.com/bytewax/bytewax/pull/139/

Reply:
Using the new testing clock now, so this is fixed